src/s2/base/port.h

```diff
@@ -58,6 +58,20 @@
 #define STDIN_FILENO 0
 #define STDOUT_FILENO 1
 #define STDERR_FILENO 2
+
+#ifdef S_IRGRP
```

Comment:
How is this defined, and why is it sometimes defined and sometimes not with MSVC?

Prefer #ifndef ... #define ... #endif all as one block below.